### PR TITLE
Added CHOP results, removed previous runs

### PR DESCRIPTION
**Pull request by GeoffNN**

### CODE REVIEW

**Comment by agramfort** [comment]
@GeoffNN results are weird https://benchopt.github.io/results/benchmark_logreg_l2_benchopt_run_2021-03-18_02h11m46.html any idea ?

**Comment by GeoffNN** [comment]
Not really...

I don't understand the COPT results either. The only ones that kind of make sense are the results for the simulated datasets...

I'll check again tomorrow

**Comment by agramfort** [comment]
there is clearly something to understand.

As we don't fit the intercept you need to check that the input data are
centered
and it should help to have features reasonably well scaled.

>

**Comment by GeoffNN** [comment]
I tried re-running by quickly applying sklearn's StandardScaler on covtype (for the CHOP methods), with no better success. Haven't had time to try more before the weekend; I'll check again next week.

**Comment by agramfort** [comment]
Could it be numerical issue with float32 in pytorch ?

**Comment by GeoffNN** [comment]
I tried with float64, same thing is going on... This is quite strange. I've never used our methods before for unconstrained opt, but never had any issues on constrained problems.

**Comment by agramfort** [comment]
well there is something to understand/debug in chop I think

>

**Comment by GeoffNN** [comment]
I just rewrote an example from scratch using chop outside of benchopt -- I uploaded the code + results in our docs [here](http://openo.pt/chop/auto_examples/plot_logistic_regression_L2_penalized.html#sphx-glr-auto-examples-plot-logistic-regression-l2-penalized-py). This one uses the StandardScaler (which is better), but the difference is definitely not enough to explain the results we were seeing...

**Comment by agramfort** [comment]
what do you suggest?

>

**Comment by GeoffNN** [comment]
I think I got it -- PyTorch requires y to be in [0, 1]. I changed self.y to be thus, so it's messes with benchopt's loss evaluation.

**Comment by agramfort** [comment]
ok great. It's an easy fix then.

>

**Comment by GeoffNN** [comment]
The issue is actually different. I get a few "infinity" values early in the optimization (steps 2 to 5 - ish) ; they actually don't hinder convergence, but benchopt picks them up and declares that the optimization diverged.

**Comment by GeoffNN** [comment]
@agramfort not sure if you saw this — catching inf values early in the optimization makes benchopt consider the procedure has diverged, even when it has not. Is this expected/wanted behavior?

**Comment by agramfort** [comment]
I don't understand how you get inf? what do you suggest it has not diverged?

**Comment by GeoffNN** [comment]
I don't quite understand the inf values either -- but there's no inf values in the iterates, and it converges to similar values as the other methods (I checked the loss vals).

**Comment by agramfort** [comment]
How do I replicate ?

**Comment by GeoffNN** [comment]
You can add `loss = np.nan_to_num(loss)` in `objective.py`

**Comment by agramfort** [comment]
Can you send me the command line you type and tell me the commit hash of your branch ?

**Comment by GeoffNN** [comment]
Found the bug: using the StandardScaler in chop.py introduced a discrepancy between the optimized objective and the objective used for evaluation in objective.py.
